### ISSUE TITLE
Excluded inline namespace broken after a5792da8

### ISSUE THREAD

**Issue by eseiler** [doxygen/doxygen#8192]
**Describe the bug**
Segfault when excluding an inline namespace:
```cpp
inline namespace foo
{
    //!\brief A struct.
    struct S{};
}
```

Default Doxyfile with: `EXCLUDE_SYMBOLS        = foo`

**Expected behavior**
No segfault.

**To Reproduce**
[example.tar.gz](https://github.com/doxygen/doxygen/files/5577378/example.tar.gz)

**Version**
a5792da8b4bf88a9d93cfd0582cec469747de3b2

**Stack trace**
```
NamespaceDefImpl::addInnerCompound(NamespaceDefImpl * const this, const Definition * d) (\run\shm\doxygen\src\namespacedef.cpp:348)
resolveClassNestingRelations() (\run\shm\doxygen\src\doxygen.cpp:1174)
parseInput() (\run\shm\doxygen\src\doxygen.cpp:11210)
main(int argc, char ** argv) (\run\shm\doxygen\src\main.cpp:37)
```

**Comment by albert-github** [comment]
Looks indeed like a missing test on `cd` in doxygen.cpp around line 1174:
```
          ClassDefMutable *cd = toClassDefMutable(icd);
          //printf("****** adding %s to scope %s in iteration %d\n",cd->name().data(),d->name().data(),iteration);
          d->addInnerCompound(cd);
```

**Comment by doxygen** [comment]
@eseiler, @albert-github Please verify if the problem is fixed with the 2 commits I just made.

**Comment by eseiler** [comment]
Works for me, thanks!

**Comment by albert-github** [comment]
Works for me also in case of b2d009555ebce4a6802ac0255c056e880bdb45bd

**Comment by albert-github** [comment]
Code has been corrected in master on github (closing issue as it only occurred in the master version).

**Comment by scpeters** [comment]
I think this fix has been released?

as an aside, I'm seeing something that may be similar in https://github.com/ignitionrobotics/ign-gazebo/issues/520 but I haven't been able to generate a simple way to reproduce it

**Comment by albert-github** [comment]
Yes, this fix has been released but slipped the automatic notification stage as it didn't have the "fixed but not released" tag anymore.

This issue was fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
